### PR TITLE
HOSS2-140 | Status found page - Header updates

### PR DESCRIPTION
**Pull request by liampan**

### CODE REVIEW

File: app/uk/gov/hmrc/homeofficeimmigrationstatus/views/StatusFoundPageContext.scala [hunk 1]
```diff
@@ -16,39 +16,36 @@
...
+  def currentStatusLabel(implicit messages: Messages): String = {
+    val prefix = "status-found.current."
+    mostRecentStatus match {
+      case Some(status) =>
+        (status.productType, status.immigrationStatus) match {
+          case (EUS, LTR)                       => messages(prefix + "EUS.LTR" + status.expiredMsg)
+          case (EUS, ILR)                       => messages(prefix + "EUS.ILR" + status.expiredMsg)
+          case (pt, LTR) if pt != EUS           => messages(prefix + "nonEUS.LTR" + status.expiredMsg)
+          case (pt, ILR) if pt != EUS           => messages(prefix + "nonEUS.ILR" + status.expiredMsg)
+          case (pt, LTE) if pt != EUS           => messages(prefix + "nonEUS.LTE" + status.expiredMsg)
```
Comment (burns238) [inline app/uk/gov/hmrc/homeofficeimmigrationstatus/views/StatusFoundPageContext.scala:41]: I'm not sure the if statements are required, right? For the first two it's covered by the fact that (EUS, xxx) was already in the match, and for LTE I think it's safe to say if we ever get LTE we display the LTE message

File: app/uk/gov/hmrc/homeofficeimmigrationstatus/models/ImmigrationStatus.scala [hunk 2]
```diff
@@ -33,15 +33,24 @@ case class ImmigrationStatus(
...
-  val hasExpired: Boolean = statusEndDate.exists(_.isBefore(LocalDate.now))
+  private val hasExpired: Boolean = statusEndDate.exists(_.isBefore(LocalDate.now))
+
+  val expiredMsg: String = if (hasExpired) ".expired" else ""
 
 }
 
 object ImmigrationStatus {
 
+  //todo proper enum?
```
Comment (burns238) [inline app/uk/gov/hmrc/homeofficeimmigrationstatus/models/ImmigrationStatus.scala:44]: Yeah it's worth creating sealed traits for ProductType and ImmigrationStatus with objects for each of these. We can always do it in a future PR if we don't want to do it here.

File: app/uk/gov/hmrc/homeofficeimmigrationstatus/views/StatusFoundPageContext.scala [hunk 3]
```diff
@@ -16,39 +16,36 @@
...
+          case (EUS, COA_IN_TIME_GRANT)         => messages(prefix + "EUS.COA_IN_TIME_GRANT" + status.expiredMsg)
+          case (EUS, POST_GRACE_PERIOD_COA)     => messages(prefix + "EUS.POST_GRACE_PERIOD_COA_GRANT" + status.expiredMsg)
+          case (FRONTIER_WORKER, PERMIT)        => messages(prefix + "FRONTIER_WORKER.PERMIT" + status.expiredMsg)
+          case (productType, immigrationStatus) => messages(prefix + "hasFBIS", productType, immigrationStatus)
```
Comment (burns238) [inline app/uk/gov/hmrc/homeofficeimmigrationstatus/views/StatusFoundPageContext.scala:45]: Do we want to write a method that both adds the prefix and expired suffix?

**Review by burns238 — APPROVED** [review]

File: app/uk/gov/hmrc/homeofficeimmigrationstatus/views/StatusFoundPageContext.scala (hunk 1 above)
Comment (liampan) [reply app/uk/gov/hmrc/homeofficeimmigrationstatus/views/StatusFoundPageContext.scala:41]: so the ticket said any non EUS, i was just playing for if EUS, PAN got introduced then it shouldnt match here and wa=ould fall in to the catch all?

File: app/uk/gov/hmrc/homeofficeimmigrationstatus/views/StatusFoundPageContext.scala (hunk 3 above)
Comment (liampan) [reply app/uk/gov/hmrc/homeofficeimmigrationstatus/views/StatusFoundPageContext.scala:45]: can do :)

File: app/uk/gov/hmrc/homeofficeimmigrationstatus/models/ImmigrationStatus.scala (hunk 2 above)
Comment (liampan) [reply app/uk/gov/hmrc/homeofficeimmigrationstatus/models/ImmigrationStatus.scala:44]: agreed. Will do in separate PR this one is already getting murky with refactoring 🙈

File: app/uk/gov/hmrc/homeofficeimmigrationstatus/views/StatusFoundPageContext.scala (hunk 1 above)
Comment (liampan) [reply app/uk/gov/hmrc/homeofficeimmigrationstatus/views/StatusFoundPageContext.scala]: oh wait see what your saying.

**Pull request merged**